server/src/main/scala/za/co/absa/atum/server/api/database/Runs.scala

```diff
@@ -84,4 +86,29 @@ object Runs {
 
     override protected def slickConverter: GetResult[Unit] = GetResult { _ => }
   }
+
+  class CreatePartitioningIfNotExists(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBSingleResultFunction[PartitioningDTO, Long, SlickPgEngine]
```

Comment:
Note: this is `Long` and not `BigInt` as in the DB, but `Long` in Scala is the same as `BigInt` in PG - 8byte signed integer.

Reply:
I was strictly following your DB function that returns `OUT id_partitioning`. But when I think about it now, we need to retrieve the original author of the partitioning so some sort of SELECT will be needed

Reply:
Regarding ID in the server, or in the client - I don't think that's needed, we can consider all IDs (except for Checkpoint ID) to be something that only DB is aware of, other pieces of the app don't care